project/ScalafixBuild.scala

```diff
@@ -311,6 +317,31 @@ object ScalafixBuild extends AutoPlugin with GhpagesKeys {
         organization.value % s"${moduleName.value}_$binaryVersion" % previousArtifactVersion
       )
     },
-    mimaBinaryIssueFilters ++= Mima.ignoredABIProblems
+    mimaBinaryIssueFilters ++= Mima.ignoredABIProblems,
+    publishLocalTransitive := Def.taskDyn {
+      val ref = thisProjectRef.value
+      publishLocal.all(ScopeFilter(inDependencies(ref)))
+    }.value,
+    crossPublishLocalBinTransitive := {
+      val currentState = state.value
+      val ref = thisProjectRef.value
+      val versions = crossScalaVersions.value
+      versions.map {
+        version =>
+          val withScalaVersion = Project
+            .extract(currentState)
+            .appendWithoutSession(
```

Comment:
with this (as opposed to `appendWithSession`), we don't observe the effect of `++`, which makes it harder to override `scalaVersion` as it seems that it's done at the project level